src/flow_solver/flow_solver_cases/dipole_wall_collision.cpp

```diff
@@ -0,0 +1,225 @@
+                unsigned int current_id = cell->face(face)->boundary_id();
+                if(this->is_oblique) {
+                    if (current_id == 2 || current_id == 3) cell->face(face)->set_boundary_id (1001); // Bottom and top wall
+                }
+                if (current_id == 0 || current_id == 1) cell->face(face)->set_boundary_id (1001); // Left and right wall
+                // could simply introduce different boundary id if using a wall model
+            }
+        }
+    }
+
+    return grid;
+}
+
+template <int dim, int nstate>
+std::vector<double> DipoleWallCollision<dim,nstate>::get_mesh_step_size_stretched(
+    const int number_of_cells_, 
+    const double domain_length_) const 
+{
+    // Set the parameters for the DWC flow case
+    const int number_of_cells_y_direction = number_of_cells_;
+    const double domain_length_y = domain_length_;
+    const double pi_val = 3.141592653589793238;
+    // ------------------------------------------------
+    // CODE BELOW TAKEN FROM CHANNEL FLOW CLASS
+    // ------------------------------------------------
+    // - get stretched spacing for y-direction to capture boundary layer
+    const int number_of_edges_y_direction = number_of_cells_y_direction+1;
+    std::vector<double> element_edges_y_direction(number_of_edges_y_direction);
+    /**
+     * Reference: C. CARTON DE WIARTET. AL, "Implicit LES of free and wall-bounded turbulent flows based onthe discontinuous Galerkin/symmetric interior penalty method", 2015.
+     **/
+    const double num_cells_y = (double)number_of_cells_y_direction;
+    const double uniform_spacing = domain_length_y/num_cells_y;
+    for (int j=0; j<(number_of_cells_y_direction/2+1); j++) {
+        element_edges_y_direction[j] = 1.0 - cos(pi_val*((double)j)*uniform_spacing/2.0);
+        element_edges_y_direction[number_of_cells_y_direction-j] = domain_length_y-element_edges_y_direction[j];
+    }
+    // - compute the step size in y-direction as the difference between element edges in y-direction
+    std::vector<double> step_size_y_direction(number_of_cells_y_direction);
+    for (int j=0; j<number_of_cells_y_direction; j++) {
+        step_size_y_direction[j] = element_edges_y_direction[j+1] - element_edges_y_direction[j];
+    }
+    return step_size_y_direction;
+}
+
+template <int dim, int nstate>
+std::shared_ptr<Triangulation> DipoleWallCollision<dim,nstate>::generate_grid_stretched() const
+{
+    // Set the parameters for the DWC flow case
+    const double domain_length_x = this->domain_right-this->domain_left;
+    const double domain_length_y = this->domain_right-this->domain_left;
+    const int number_of_cells_x_direction = this->number_of_cells_per_direction;
+    const int number_of_cells_y_direction = this->number_of_cells_per_direction;
+    // ------------------------------------------------
+    // CODE BELOW TAKEN FROM CHANNEL FLOW CLASS -- Commented z-component stuff since 2D here
```

Comment:
Could you move the grid generation here and in channel flow to the `src/mesh/grids` directory and merge this with `turbulent_channel_flow`? Might be some additional work to generalize dimensions, stretch factor, which face is refined etc but it could be useful for the future.

Reply:
I thought of that, however these mesh stretching functions are specific to the turbulent channel flow / dipole case so I chose to leave them in here; I think could be done by in a future PR by someone else that wants to benefit from this meshing code and use it in their case